src/bindings/python/src/openvino/frontend/pytorch/module_extension.py

```diff
@@ -0,0 +1,42 @@
+
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+import typing
+
+class ModuleExtension:
+    def __init__(self, module, target_op, evaluate=None, convert=None):
+        """
+        Creates an extension that replaces entire PyTorch module by a single operation.
+        This functionality works with PyTorch models only. A module can be indentified by
+        module type (e.g. torch.nn.Linear), module instance in the model or module name.
+
+        Args:
+            module (str, torch.nn.Module, type(torch.nn.Module)): PyTorch module to replace
+
+            target_op (str): a target operation that will be used as a replacer for the module,
```

Comment:
what if I want to decompose existing `torch.nn.Module` into a set of operations? what is `target_op` here?

Reply:
Only one custom operation is supported. It will be substituted instead of the target_op argument. It extend it furhter, I think we can give access to all custom/existing operations passing a kind of builder instead of a single target_op. It would be a more scalable approach, but I haven't drafted it yet. It wouldn't use `target_op` parameter in ModuleExtension.__init__, and in `convert` parameter you will write something like ops.PagedAttentionExtension(args), where `ops` is an op builder, and PageAttentionExtension is the name of the custom op, so you would able to instantiate many ops of different type. It is not so straightforward, because evaluate is still defined at the level of each op, but we can experiment with that approach.